src/python/pants/core/goals/fmt.py

```diff
@@ -125,20 +134,13 @@ class FmtRequest(LintRequest):
     @frozen_after_init
     @dataclass(unsafe_hash=True)
     class SubPartition(LintRequest.SubPartition):
-        _snapshot: Snapshot | None = None
+        snapshot: Snapshot
 
         @property
         def files(self) -> tuple[str, ...]:
             return self.elements
```

Comment:
This could now delegate to `self.snapshot.files` instead.

Reply:
I'm really pained by the duplication here between `elements` and `snapshot.files`. At the end of the day, as long as the user isn't pained by the duplication I can sleep at night. 

As discussed privately, I think there's a world of solutions where the snapshot is a superset of the elements, so I prefer (lightly) using `elements` here.